src/services/exit_order.py

```diff
@@ -1,24 +1,286 @@
+        module_operator_ids = {}
+        for module_id, operator_id in operator_validators:
+            module_operator_ids.setdefault(module_id, set()).add(operator_id)
+
+        last_requested_to_exit_indexes = {}
+        for module_id, operator_ids in module_operator_ids.items():
+            operator_ids.sort()
+            per_operator_indexes = self._get_last_requested_validator_index(blockstamp, module_id, operator_ids)
+            for operator_id in operator_ids:
+                last_requested_to_exit_indexes[(module_id, operator_id)] = per_operator_indexes[operator_id]
+        return last_requested_to_exit_indexes
+
+    def _get_delayed_validators_per_operator(
+        self,
+        blockstamp: BlockStamp,
+        operator_validators: dict[NodeOperatorIndex, list[LidoValidator]],
+        last_requested_to_exit_indices_per_operator: dict[NodeOperatorIndex, int],
+    ) -> dict[NodeOperatorIndex, int]:
+
+        delayed_validators_per_operator = defaultdict(int)
+
+        # We don't consider validator as delayed if it was requested to exit
+        # in last VALIDATOR_DELAYED_TIMEOUT_IN_SLOTS slots
+        recently_requested_to_exit_indices_per_operator = self._get_recently_requested_to_exit_indices(
+            blockstamp, operator_validators
+        )
+
+        for module_operator, validators in operator_validators.items():
+            recently_requested_to_exit_indices = recently_requested_to_exit_indices_per_operator[module_operator]
+            for validator in validators:
+                previously_requested_to_exit = (
+                    int(validator.index) <= last_requested_to_exit_indices_per_operator[module_operator]
+                )
+                on_exit = self._is_on_exit(validator)
+                recently_requested_to_exit = int(validator.index) in recently_requested_to_exit_indices
+                if previously_requested_to_exit and not on_exit and not recently_requested_to_exit:
+                    delayed_validators_per_operator[module_operator] += 1
+
+        return delayed_validators_per_operator
+
+    def _get_recently_requested_to_exit_indices(
+        self, blockstamp: BlockStamp, operator_validators: dict[NodeOperatorIndex, list[LidoValidator]]
+    ) -> dict[NodeOperatorIndex, set[int]]:
+        """
+        Returns recently requested to exit validators indices per operator
+        """
+        module_operator = {}
+
+        # Initialize dict with empty sets for operators which validators were not contained in any event
+        for operator in operator_validators:
+            module_operator[operator] = set()
+
+        from_block = blockstamp.ref_slot - VALIDATOR_DELAYED_TIMEOUT_IN_SLOTS
+        to_block = blockstamp.ref_slot
+        events = self.w3.lido_contracts.validators_exit_bus_oracle.events.ValidatorExitRequest.getLogs(
+            fromBlock=from_block, toBlock=to_block
+        )
+        from_block_timestamp = from_block * self.c_conf.slots_per_epoch
```

Comment:
I guess it should look smth like this: 
```python
real_range_border_timestamp = get_timestamp_from_slot(blockstamp.ref_slot - VALIDATOR_DELAYED_TIMEOUT_IN_SLOTS)

def get_timestamp_from_slot(slot):
  return slot * seconds_per_slot + genesis_time
```

Reply:
yep. I just forgot to add `genesis_time`. fixed

Reply:
there are a couple more issues here:
- need to calc it from `ref_slot` since here we are interested in a time range: `blockstamp.ref_slot - VALIDATOR_DELAYED_TIMEOUT_IN_SLOTS`. Blocks are not slots and can't be used to calc time ranges
- `slots_per_epoch` need to be replace with `seconds_per_slots`

I think it's worth detailing here what's going on. it's not an obvious thing that needs to be documented

Reply:
fixed. Added couple words about it